### PR TITLE
EWPP-47: List page selection in plugin.

### PR DESCRIPTION
**Pull request by hernani**
### Description

[Insert description here]

### Change log

- Added:
- Changed:
- Deprecated:
- Removed:
- Fixed:
- Security:

### Commands

```sh
[Insert commands here]

```

### CODE REVIEW

File: src/ListPresetFiltersBuilder.php [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+   *   The built form.
+   */
+  protected function buildEditPresetFilter(array $form, FormStateInterface $form_state, string $form_key, ListSourceInterface $list_source, array $available_filters, array $current_filters, string $filter_key) {
+    $form[$form_key]['preset_filters_wrapper']['edit'] = [
+      '#type' => 'fieldset',
+      '#title' => $this->t('Set default value for :filter', [':filter' => $available_filters[$filter_key]]),
+    ];
+
+    // Execute list source query so we have results.
+    $list_source->getQuery()->execute();
```
Comment (upchuk) [inline src/ListPresetFiltersBuilder.php:244]: Why?

File: src/Plugin/facets/widget/MultiselectWidget.php [hunk 2]
```diff
@@ -16,7 +24,127 @@
...
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildDefaultValuesWidget(FacetInterface $facet, ListSourceInterface $list_source = NULL, array $parents = []): ?array {
+    $field_definition = $this->getFieldDefinition($facet, $list_source);
+    $field_type = !empty($field_definition) ? $field_definition->getType() : NULL;
+
+    if (in_array(EntityReferenceFieldItemListInterface::class, class_implements($field_definition->getClass()))) {
+      for ($i = 0; $i < count($facet->getActiveItems()) + 1; $i++) {
```
Comment (upchuk) [inline src/Plugin/facets/widget/MultiselectWidget.php:113]: Why are we counting the active items here? Why do we need to rely on the facets for building these form elements?

**Review by upchuk — CHANGES_REQUESTED** [review]

File: src/Plugin/facets/widget/MultiselectWidget.php (hunk 2 above)
Comment (hernani) [reply src/Plugin/facets/widget/MultiselectWidget.php:113]: The facet have the list of saved values and we count the list of active values for the edition of multiple values allowing also to add a new default value anytime the value is edited.

**Review by imanoleguskiza — APPROVED** [review]

**Review by upchuk — APPROVED** [review]

**Pull request merged**